### PR TITLE
Improved release notes generation

### PR DESCRIPTION
**Pull request by szczepiq**
Fixes issue #598

Does not change Mockito library code, just the release pipeline.

### CODE REVIEW

File: buildSrc/src/test/groovy/org/mockito/release/notes/vcs/DefaultContributionSetTest.groovy [hunk 1]
```diff
@@ -49,6 +49,35 @@ class DefaultContributionSetTest extends Specification {
...
+        contributions.add(new GitCommit("B@B", "B", ""))
+        contributions.add(new GitCommit("a@a", "a", ""))
+
+        expect:
+        contributions.toText() == """* Authors: 4
+* Commits: 5
+  * 2: d
+  * 1: a
+  * 1: B
+  * 1: c"""
```
Comment (mkordas) [inline buildSrc/src/test/groovy/org/mockito/release/notes/vcs/DefaultContributionSetTest.groovy]: @szczepiq you could use [`stripIndent`](http://mrhaki.blogspot.com/2010/06/groovy-goodness-strip-leading-spaces.html) for better readability here

**Comment by TimvdLippe** [comment]
How can I inspect the changed results? The changes SGTM, but would like to see the output.

File: buildSrc/src/main/groovy/org/mockito/release/notes/vcs/DefaultContributionSet.java [hunk 2]
```diff
@@ -20,12 +22,29 @@ void add(Commit commit) {
...
+
+    private static Contribution findContribution(Commit commit, Iterable<Contribution> contributions) {
+        for (Contribution c : contributions) {
+            //From Git Log we don't know the GitHub user ID, only the email and name.
+            //Sometimes contributors have different email addresses while the same name
+            //This leads to awkward looking release notes, where same author is shown multiple times
+            //We consider the contribution to be the same if any of: email or name is the same
+            //
+            //This approach comes with a caveat. What if the user have same author name, different email and indeed it is a different user?
+            // This scenario is not handled well but it is unlikely and we consider it a trade-off
```
Comment (lukasniemeier-zalando) [inline buildSrc/src/main/groovy/org/mockito/release/notes/vcs/DefaultContributionSet.java]: check [out](https://git-scm.com/docs/git-check-mailmap) out git's `.mailmap`, might help in this case

Comment (szczepiq) [inline buildSrc/src/main/groovy/org/mockito/release/notes/vcs/DefaultContributionSet.java]: Thanks for hint and review!!!

I didn't know about this command. I've tried it but it does not give me github user id. However, it does seem interesting and it solves the problem of misspelled names / emails. Wanna help, PR! :)

File: buildSrc/src/test/groovy/org/mockito/release/notes/vcs/DefaultContributionSetTest.groovy (hunk 1 above)
Comment (szczepiq) [inline buildSrc/src/test/groovy/org/mockito/release/notes/vcs/DefaultContributionSetTest.groovy]: I'll give it a shot next time. Thanks for the tip!!!

I'm so used to used to just multi-line string that my brain does not even register that it is hard to read ;)

**Comment by szczepiq** [comment]
@TimvdLippe, I apologize for not documenting the release procedure hints well enough (or make it discoverable enough). There Gradle tasks that enable to preview the release notes (see gradle tasks output)

Steps to preview release notes for RC.1:
 - delete all content from current release notes file up to the previous stable version
 - ./gradlew previewReleaseNotes

**Comment by szczepiq** [comment]
I'm merging this PR if that's ok. I'd like to make some progress on the next item (grouping for improvements, it used to work :D)

**Comment by szczepiq** [comment]
Thanks everybody for review!!! It's really energizing that you follow the work on Mockito and give juicy implementation hints.

**Pull request merged**